tests/base-notebook/test_healthcheck.py

```diff
@@ -13,23 +13,106 @@
 
 
 @pytest.mark.parametrize(
-    "env",
+    "env,cmd,user",
     [
-        None,
-        ["DOCKER_STACKS_JUPYTER_CMD=lab"],
-        ["DOCKER_STACKS_JUPYTER_CMD=notebook"],
-        ["DOCKER_STACKS_JUPYTER_CMD=server"],
-        ["DOCKER_STACKS_JUPYTER_CMD=nbclassic"],
-        ["RESTARTABLE=yes"],
-        ["JUPYTER_PORT=8171"],
-        ["JUPYTER_PORT=8117", "DOCKER_STACKS_JUPYTER_CMD=notebook"],
+        (None, None, None),
+        (["DOCKER_STACKS_JUPYTER_CMD=lab"], None, None),
+        (["DOCKER_STACKS_JUPYTER_CMD=notebook"], None, None),
+        (["DOCKER_STACKS_JUPYTER_CMD=server"], None, None),
+        (["DOCKER_STACKS_JUPYTER_CMD=nbclassic"], None, None),
+        (["RESTARTABLE=yes"], None, None),
+        (["JUPYTER_PORT=8171"], None, None),
+        (["JUPYTER_PORT=8117", "DOCKER_STACKS_JUPYTER_CMD=notebook"], None, None),
+        (None, ["start-notebook.sh", "--NotebookApp.base_url=/test"], None),
+        (None, ["start-notebook.sh", "--NotebookApp.base_url=/test/"], None),
+        (["GEN_CERT=1"], ["start-notebook.sh", "--NotebookApp.base_url=/test"], None),
+        (
+            ["GEN_CERT=1", "JUPYTER_PORT=7891"],
+            ["start-notebook.sh", "--NotebookApp.base_url=/test"],
+            None,
+        ),
+        (["NB_USER=testuser", "CHOWN_HOME=1"], None, "root"),
+        (
+            ["NB_USER=testuser", "CHOWN_HOME=1"],
+            ["start-notebook.sh", "--NotebookApp.base_url=/test"],
+            "root",
+        ),
+        (
+            ["NB_USER=testuser", "CHOWN_HOME=1", "JUPYTER_PORT=8123"],
+            ["start-notebook.sh", "--NotebookApp.base_url=/test"],
+            "root",
+        ),
     ],
 )
-def test_health(container: TrackedContainer, env: Optional[list[str]]) -> None:
+def test_health(
+    container: TrackedContainer,
+    env: Optional[list[str]],
+    cmd: Optional[list[str]],
+    user: Optional[str],
+) -> None:
     running_container = container.run_detached(
         tty=True,
         environment=env,
+        command=cmd,
+        user=user,
     )
+
     # sleeping some time to let the server start
-    time.sleep(15)
+    time_spent = 0.0
+    wait_time = 0.1
+    time_limit = 15
+    while time_spent < time_limit:
+        time.sleep(wait_time)
+        time_spent += wait_time
+        if get_health(running_container) == "healthy":
+            return
+
     assert get_health(running_container) == "healthy"
+
+
+@pytest.mark.parametrize(
+    "env,cmd,user",
+    [
+        (["NB_USER=testuser", "CHOWN_HOME=1"], None, None),
+        (["NB_USER='test user'", "CHOWN_HOME=1"], None, "root"),
```

Comment:
This should actually work.
I think it's better to fix users with spaces after merging this PR.

Reply:
Why do you think it should work? I think it's a limitation of Unix in general, you just can't have a username that contain spaces.

Reply:
I'm not really sure though. That's just my impression when reading the error message.

Reply:
Sorry you're right. At least based on https://serverfault.com/a/73101 and useradd's help, it should be possible to allow space, by giving the `--badnames` param to the `useradd` command. But it's an entirely different issue I guess, as you said. I'll leave it to you whether to address this issue and actually allow "bad username". 

Should we remove these tests, then?

Reply:
You're right.

```
# useradd "crazy user"
useradd: invalid user name 'crazy user'
```

I will delete these tests then (they don't make much sense I suppose).